bento_wes/events.py

```diff
@@ -1,30 +1,123 @@
-from bento_lib.events import EventBus, types as et
-from flask import g
+from __future__ import annotations
 
-from .config import BENTO_EVENT_REDIS_URL
+from typing import Annotated, Awaitable, Callable, Optional
+import os
+from fastapi import Depends
+from bento_lib.events import EventBus, types as et
 
+from .config import get_settings
+from .logger import logger
 
 __all__ = [
-    "get_new_event_bus",
-    "get_flask_event_bus",
-    "close_flask_event_bus",
+    "_create_event_bus",
+    "init_event_bus",
+    "shutdown_event_bus",
+    "get_event_bus",
+    "EventBusDep",
 ]
 
+# ---------- Singleton state ----------
+_BUS: Optional[EventBus] = None
+
+
+# ---------- Construction ----------
+def _create_event_bus() -> EventBus:
+    """
+    Create and configure the EventBus instance (no I/O side-effects here).
+    """
+    settings = get_settings()
+    bus = EventBus(url=settings.bento_event_redis_url, allow_fake=True, logger=logger)
+
+    # Register all event types here
+    bus.register_service_event_type(et.EVENT_WES_RUN_UPDATED, et.EVENT_WES_RUN_UPDATED_SCHEMA)
+    bus.register_service_event_type(et.EVENT_WES_RUN_FINISHED, et.EVENT_WES_RUN_FINISHED_SCHEMA)
+    bus.register_service_event_type(et.EVENT_CREATE_NOTIFICATION, et.EVENT_CREATE_NOTIFICATION_SCHEMA)
+    return bus
+
+
+async def _close_event_bus(bus: EventBus) -> None:
+    """
+    Gracefully close the bus whether it exposes `aclose()` or `close()`.
```

Comment:
where does this come from? there is no close or aclose function on the eventbus class... read the class to get an idea of how it works.